src/amo/components/HeroRecommendation/index.js

```diff
@@ -66,58 +75,76 @@ export class HeroRecommendationBase extends React.Component<InternalProps> {
   };
 
   render() {
-    const { _checkInternalURL, i18n, shelfData } = this.props;
-    const { addon, description, external, gradient, featuredImage } = shelfData;
-
-    const linkInsides = <span> {i18n.gettext('Get the extension')} </span>;
-
+    const {
+      _checkInternalURL,
+      i18n,
+      shelfData,
+      siteIsReadOnly,
+      siteNotice,
+    } = this.props;
+    const { addon, description, external, featuredImage, gradient } =
+      shelfData || {};
+
+    let gradientsClassName;
     let heading;
     let link;
 
-    const linkProps = _checkInternalURL({
-      urlString: this.makeCallToActionURL(),
-    }).isInternal
-      ? {}
-      : { rel: 'noopener noreferrer', target: '_blank' };
+    const heightClassName =
+      siteIsReadOnly || siteNotice
+        ? 'HeroRecommendation--height-with-notice'
```

Comment:
Why do you need to adjust the height for notices? Could we just let the notice push the content down when it exists?

Reply:
That doesn't work, although this does point out a problem with the sizing for when a notice exists. We need the `min-height` to include the notice otherwise on the skeleton version of the page it doesn't make the primary hero area large enough. This does mean, however, that the size is currently designed to use the exact size of the notice as it currently is on dev, and there's no reason to expect it will always be that size. 😢 

I'm not sure what a good solution to this is. On the bright side, I don't think we expect for there to be notices on the site very often, so if there is a bit of jumpiness between the skeleton and the final screen when a notice does exist, that shouldn't have a huge impact.

I do think we need to figure out a better way to deal with this, although it might suffer from the same problem as the hero element itself, which is that we can never be sure exactly what size it's going to be, and can only make a best guess at what size it is likely to be. I'll open a follow-up so that we can put some thought into a solution for this.

Reply:
I opened #8727 as a follow-up for this.